Fancy Extracter Plugin

ray-hc opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/ray-hc/obsidian-fancy-extracter

## Release Checklist
- [X] I have tested the plugin on
  - [ ]  Windows
  - [X]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [X] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [X] `main.js`
  - [X] `manifest.json`
  - [X] `styles.css` _(optional)_
- [X] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [X] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [X] My README.md describes the plugin's purpose and provides clear usage instructions.
- [X] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [X] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [X] I have added a license in the LICENSE file.
- [X] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

Zachatoo commented:
["description": "Plugin which allows users to extract notes to subfolders and use auto-generated names.",](https://github.com/ray-hc/obsidian-fancy-extractor/blob/198b4091390352412bade30ee676b36bb1dcff92/manifest.json#L6)
Avoid including sentences like "Plugin which allows ..." in your description, it should be a given since this is only shown when browsing plugins through the app or on our website.
We have published a guide for plugin descriptions to follow here: https://docs.obsidian.md/Plugins/Releasing/Submission+requirements+for+plugins#Keep+plugin+descriptions+short+and+simple.

[id: 'fancy-extract-open-modal',](https://github.com/ray-hc/obsidian-fancy-extractor/blob/198b4091390352412bade30ee676b36bb1dcff92/main.ts#L19), [id: 'fancy-extract-use-default',](https://github.com/ray-hc/obsidian-fancy-extractor/blob/198b4091390352412bade30ee676b36bb1dcff92/main.ts#L24)
Don't prefix the command id with the plugin id, Obsidian will make sure that there are no conflicts with other plugins.

[name: 'Extract (Open Name Modal)',](https://github.com/ray-hc/obsidian-fancy-extractor/blob/198b4091390352412bade30ee676b36bb1dcff92/main.ts#L20), [name: 'Extract (Use Default Name)',](https://github.com/ray-hc/obsidian-fancy-extractor/blob/198b4091390352412bade30ee676b36bb1dcff92/main.ts#L25), [contentEl.createEl('h1', { text: 'Naming Extracted Note' });](https://github.com/ray-hc/obsidian-fancy-extractor/blob/198b4091390352412bade30ee676b36bb1dcff92/main.ts#L132), [.setName('Subfolder Name')](https://github.com/ray-hc/obsidian-fancy-extractor/blob/198b4091390352412bade30ee676b36bb1dcff92/main.ts#L180), [.setName('Use Subfolder')](https://github.com/ray-hc/obsidian-fancy-extractor/blob/198b4091390352412bade30ee676b36bb1dcff92/main.ts#L191), [.setName("Text After Extraction")](https://github.com/ray-hc/obsidian-fancy-extractor/blob/198b4091390352412bade30ee676b36bb1dcff92/main.ts#L202), [new Setting(containerEl).setName('Default Note Name').setHeading();](https://github.com/ray-hc/obsidian-fancy-extractor/blob/198b4091390352412bade30ee676b36bb1dcff92/main.ts#L217), [.setName("First N Words")](https://github.com/ray-hc/obsidian-fancy-extractor/blob/198b4091390352412bade30ee676b36bb1dcff92/main.ts#L230), [.setName('Custom Words to Filter')](https://github.com/ray-hc/obsidian-fancy-extractor/blob/198b4091390352412bade30ee676b36bb1dcff92/main.ts#L247)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[editorCallback: (editor, view: MarkdownView) => this.extractText(editor, view, true)](https://github.com/ray-hc/obsidian-fancy-extractor/blob/198b4091390352412bade30ee676b36bb1dcff92/main.ts#L21), [editorCallback: (editor, view: MarkdownView) => this.extractText(editor, view, false)](https://github.com/ray-hc/obsidian-fancy-extractor/blob/198b4091390352412bade30ee676b36bb1dcff92/main.ts#L26)
These should be a [editor check commands](https://docs.obsidian.md/Plugins/User+interface/Commands#Editor+commands), should be a selection before allowing the command to be run.

[new Setting(containerEl)](https://github.com/ray-hc/obsidian-fancy-extractor/blob/198b4091390352412bade30ee676b36bb1dcff92/main.ts#L190-L200)
If a setting does not apply when another setting is toggled on/off, then do not show that setting, only show settings that are relevant. You can conditionally show/hide relevant settings like this.
```js
new Setting(containerEl)
    .setName('Toggle')
    .addToggle((toggle) =>
        toggle
            .setValue(this.plugin.settings.toggle)
            .onChange(async (value) => {
                this.plugin.settings.toggle = value;
                await this.plugin.saveSettings();
                // Re-run display function to show/hide related settings
                this.display();
            })
    );
if (this.plugin.settings.toggle) {
    new Setting(containerEl)
        .setName('Value')
        .setDesc('Should only be visible when Toggle is on')
        .addText((text) =>
            text
                .setValue(this.plugin.settings.value)
                .onChange(async (value) => {
                    this.plugin.settings.value = value;
                    await this.plugin.saveSettings();
                })
        );
}
```

ray-hc commented:
1) Fixed description
2) Updated ID. I see the plugin guidelines on command ID now, I was going off the docs [here](https://docs.obsidian.md/Reference/TypeScript+API/Command/id) which recommend "Globally unique ID"
3) Fixed sentence case
4) Converted to editor check callback
5) Removed the setting in question which is conditional, which should hopefully make user options more clear

joethei commented:
[name: 'Extract (Open Name Modal)',](https://github.com/ray-hc/obsidian-fancy-extractor/blob/7e709d38dfe0f84969019dacc47cff77a6f10cc5/main.ts#L22), [name: 'Extract (Use Default Name)',](https://github.com/ray-hc/obsidian-fancy-extractor/blob/7e709d38dfe0f84969019dacc47cff77a6f10cc5/main.ts#L29), [contentEl.createEl('h1', { text: 'Naming Extracted Note' });](https://github.com/ray-hc/obsidian-fancy-extractor/blob/7e709d38dfe0f84969019dacc47cff77a6f10cc5/main.ts#L152)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)
